test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+	cmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- ` + args
+	logrus.Debugf("Exec: Running command -- %s", cmd)
+	out, err := k8sMaster.runCommand(cmd)
+	if err != nil {
+		return out, err
+	}
+
+	return out, nil
+}
+
+/*
+* execBG executes a background process on the node
+ */
+func (k *kubePod) execBG(podName, args string, ns ...string) {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+	cmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- ` + args
+	logrus.Debugf("ExecBG: Running command -- %s", cmd)
+	k8sMaster.tbnode.RunCommandBackground(cmd)
+}
+
+/*
+* podExec function is used to run a command typically multiple commands
+* with pipes and redirections within the pod rather than the node
+ */
+func (k *kubePod) podExec(podName, args string, ns ...string) (string, error) {
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	// Backticks and quotes here ensure that the command
+	// is properly wrapped to execute on pod rather than on the node
+	podCmd := `sh -c '\''` + args + `'\''`
+	return k.exec(podName, podCmd, ns...)
+}
+
+/*
+* podExecBG function is used to run a background command
+* within the pod rather than the node
+ */
+func (k *kubePod) podExecBG(podName, args string, ns ...string) error {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	//  - Backticks and quotes here ensure that the command
+	//    is properly wrapped to execute on pod rather than on the node
+	//  - nohup along with & ensures that the process continues to live
+	//    after the shell is terminated
+	// Since the command is to run in BG in the pod and not on the node,
+	// RunCommandBackground is not required here
+
+	podCmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- nohup sh -c '\''` + args + ` &'\''`
+	logrus.Debugf("Pod Exec BG: Running command -- %s", podCmd)
+	return k8sMaster.tbnode.RunCommand(podCmd)
+}
+
+func (k *kubePod) start(c *container) error {
+	//Kubernetes does not support start/stop
+	return nil
+}
+
+func (k *kubePod) stop(c *container) error {
+	//Kubernetes does not support start/stop
+	return nil
+}
+
+func (k *kubePod) rm(c *container) error {
+	logrus.Infof("Removing Pod: %s on %s", c.containerID, c.node.Name())
+	k8sMaster.tbnode.RunCommand(fmt.Sprintf("kubectl delete pod %s", c.name))
+	for i := 0; i < 80; i++ {
+		out, _ := k8sMaster.tbnode.RunCommandWithOutput(fmt.Sprintf("kubectl get pod %s", c.containerID))
+		if strings.Contains(out, "not found") {
+			return nil
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return fmt.Errorf("Error Termininating pod %s on node %s", c.name, c.node.Name())
+}
+
+func (k *kubePod) startListener(c *container, port int, protocol string) error {
```

Comment:
Will continue to use this for now. If you feel a strong case to change this, will open another PR. Have to change it  in all modes and don't want to take the focus off this PR.